compiler/rustc_hir_typeck/src/method/mod.rs

```diff
@@ -577,9 +569,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
 
         let pick = self.probe_for_name(
-            span,
             probe::Mode::Path,
-            method_name,
+            method_name.with_span_pos(span),
```

Comment:
can you add a comment explaining why `with_span_pos` is used here? That feels weird.

Do we get weird error messages if we change this to use the old `method_name` span directly?